### PR TITLE
Create microsoft-edge

### PR DESCRIPTION
**Pull request by robertleeblairjr**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by peterlewis** [comment]
Howdy! Hah! Looks like you beat me by one minute... ;)

Just a couple of points:

1. Your appcast is pointing towards the beta release
2. When uninstalling, I'm not sure if removing the whole of `/Library/Application Support/Microsoft` is a good idea. For example, on my system, having just installed Microsoft Edge that wouldn't remove anything microsoft-edge-related files but would trash files relating to MAU2.0 (Microsoft AutoUpdate) and TeamsUpdaterDaemon (Microsoft Teams).

If you want to reference an example of the points above you can find them at #75645. I'll leave it open for now, just in case, but you may wish to take the above into account, in your pull request. Note I've not included 'zap' removals in my request, at this stage as again, I'm not sure of how much of a good idea it is to remove Microsoft AutoUpdate... :)

Thanks!

Peter :)

**Comment by robertleeblairjr** [comment]
> Howdy! Hah! Looks like you beat me by one minute... ;)
> 
> Just a couple of points:
> 
> 1. Your appcast is pointing towards the beta release
> 2. When uninstalling, I'm not sure if removing the whole of `/Library/Application Support/Microsoft` is a good idea. For example, on my system, having just installed Microsoft Edge that wouldn't remove anything microsoft-edge-related files but would trash files relating to MAU2.0 (Microsoft AutoUpdate) and TeamsUpdaterDaemon (Microsoft Teams).
> 
> If you want to reference an example of the points above you can find them at #75645. I'll leave it open for now, just in case, but you may wish to take the above into account, in your pull request. Note I've not included 'zap' removals in my request, at this stage as again, I'm not sure of how much of a good idea it is to remove Microsoft AutoUpdate... :)
> 
> Thanks!
> 
> Peter :)

Thanks. I can close it or just add/edit it. I just copied the template from the previous betas since there wouldn't have been much of a change other than naming conventions. I'll try to find the current appcast for stable or someone can make a pull request to change the appcast if this one is accepted.

**Comment by peterlewis** [comment]
Feel free to edit yours and I'll close mine! :) I'll make some suggestions on this pull request and close mine, now. Thanks! :)

File: Casks/microsoft-edge [hunk 1]
```diff
@@ -0,0 +1,30 @@
+cask 'microsoft-edge' do
+  version '79.0.309.65'
+  sha256 '67b1e8e036c575782b1c9188dd48fa94d9eabcb81947c8632fd4acac7b01644b'
+
+  # officecdn-microsoft-com.akamaized.net was verified as official when first introduced to the cask
+  url "https://officecdn-microsoft-com.akamaized.net/pr/C1297A47-86C4-4C1F-97FA-950631F94777/MacAutoupdate/MicrosoftEdge-#{version}.pkg"
+  appcast 'https://officecdn-microsoft-com.akamaized.net/pr/C1297A47-86C4-4C1F-97FA-950631F94777/MacAutoupdate/0409EDBT01-chk.xml',
```
Comment (peterlewis) [inline Casks/microsoft-edge]: It appears to me that the correct appcast URL is `https://officecdn-microsoft-com.akamaized.net/pr/C1297A47-86C4-4C1F-97FA-950631F94777/MacAutoupdate/0409EDGE01-chk.xml`

File: Casks/microsoft-edge [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          configuration: "#{version.major}.#{version.patch}"
+  name 'Microsoft Edge'
+  homepage 'https://www.microsoft.com/en-us/edge'
+
+  auto_updates true
+
+  pkg "MicrosoftEdge-#{version}.pkg"
+
+  uninstall pkgutil: 'com.microsoft.edgemac',
+            rmdir:   '/Library/Application Support/Microsoft'
```
Comment (peterlewis) [inline Casks/microsoft-edge]: Not so sure about deleting `/Library/Application Support/Microsoft`. As mentioned, when uninstalling, I'm not sure if removing the whole of /Library/Application Support/Microsoft is a good idea. For example, on my system, having just installed Microsoft Edge that wouldn't remove anything microsoft-edge-related files but would trash files relating to MAU2.0 (Microsoft AutoUpdate) and TeamsUpdaterDaemon (Microsoft Teams).

File: Casks/microsoft-edge [hunk 3]
```diff
@@ -0,0 +1,30 @@
+cask 'microsoft-edge' do
+  version '79.0.309.65'
+  sha256 '67b1e8e036c575782b1c9188dd48fa94d9eabcb81947c8632fd4acac7b01644b'
+
+  # officecdn-microsoft-com.akamaized.net was verified as official when first introduced to the cask
+  url "https://officecdn-microsoft-com.akamaized.net/pr/C1297A47-86C4-4C1F-97FA-950631F94777/MacAutoupdate/MicrosoftEdge-#{version}.pkg"
+  appcast 'https://officecdn-microsoft-com.akamaized.net/pr/C1297A47-86C4-4C1F-97FA-950631F94777/MacAutoupdate/0409EDBT01-chk.xml',
+          configuration: "#{version.major}.#{version.patch}"
+  name 'Microsoft Edge'
+  homepage 'https://www.microsoft.com/en-us/edge'
```
Comment (peterlewis) [inline Casks/microsoft-edge]: Not sure as to what is correct, but just for your information, in my PR I used the language-agnostic link of `https://www.microsoftedge.com/` - As I say, I'm not sure what's correct, so feel free to disregard this, if it's not needed! :)

File: Casks/microsoft-edge [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  auto_updates true
+
+  pkg "MicrosoftEdge-#{version}.pkg"
+
+  uninstall pkgutil: 'com.microsoft.edgemac',
+            rmdir:   '/Library/Application Support/Microsoft'
+
+  zap launchctl: [
```
Comment (peterlewis) [inline Casks/microsoft-edge]: As I mentioned, I've not included anything 'zap'-related in mine, as I felt uneasy about including stuff that would rip-out Microsoft AutoUpdate. Maybe that's the whole point of 'zap' but thought it's a distinct possibility that someone would want to uninstall Microsoft Edge whilst leaving Microsoft Office intact! Would welcome feedback here...

**Review by peterlewis — CHANGES_REQUESTED** [review]
A few suggestions. Hope this doesn't make me sound like a know-it-all, I most definitely don't, I just noticed a few differences between our PRs so though I'd mention them! Thanks! :)

**Review by robertleeblairjr — COMMENTED** [review]
Thanks. Give me time and I'll change to your suggestions. I'm busy with other things, as well.

**Comment by peterlewis** [comment]
> Thanks. Give me time and I'll change to your suggestions. I'm busy with other things, as well.

Thanks for taking the time to consider the changes! As I say, I'm not sure about the removal of 'zap', it was just a bit more than I was comfortable with, with my own PR.

File: Casks/microsoft-edge (hunk 4 above)
Comment (robertleeblairjr) [reply Casks/microsoft-edge]: I think zap would be fine if it's definitely Edge related files. I agree with using it for MAU or other common, universal Microsoft related files.

File: Casks/microsoft-edge (hunk 3 above)
Comment (robertleeblairjr) [reply Casks/microsoft-edge]: Thanks. I just removed the language specific.

File: Casks/microsoft-edge (hunk 2 above)
Comment (robertleeblairjr) [reply Casks/microsoft-edge]: Removed it. I appreciate the suggestions.

**Comment by robertleeblairjr** [comment]
> > Thanks. Give me time and I'll change to your suggestions. I'm busy with other things, as well.
> 
> Thanks for taking the time to consider the changes! As I say, I'm not sure about the removal of 'zap', it was just a bit more than I was comfortable with, with my own PR.

I appreciate your pointers. They were very helpful and I merely did a copy/paste from the previous template without going over the details concerning MAU. I agree that the MAU related files should probably stay. Let me know if you notice anything else that may need to be added/removed.

**Comment by peterlewis** [comment]
Thank you for taking them into account! Interestingly, though I've just had a look at:

https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md

This states two interesting points:

1. User-specific preferences are removed using 'zap' (i.e. the .plist files that you're currently including with the uninstall)
2. It specifically mentions "Shared resources such as application updaters. Since shared resources may be removed, other applications may be affected by `brew cask zap`. Understanding that is the responsibility of the end user."

Based on the above, do you think that including MAU in the zap (along with the user preferences) might be the right decision, afterall? Sorry for the change in tune, I'm learning as I contribute! :)

**Review by robertleeblairjr — COMMENTED** [review]
Thanks to @peterlewis for the suggestions for the cask.

**Comment by robertleeblairjr** [comment]
> Thank you for taking them into account! Interestingly, though I've just had a look at:
> 
> https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md
> 
> This states two interesting points:
> 
> 1. User-specific preferences are removed using 'zap' (i.e. the .plist files that you're currently including with the uninstall)
> 2. It specifically mentions "Shared resources such as application updaters. Since shared resources may be removed, other applications may be affected by `brew cask zap`. Understanding that is the responsibility of the end user."
> 
> Based on the above, do you think that including MAU in the zap (along with the user preferences) might be the right decision, afterall? Sorry for the change in tune, I'm learning as I contribute! :)

I didn't really consider it since I took all of a couple of minutes to submit a copy/paste. After your suggestion about MAU, then I agreed about the problem that could occur because other casks which would install and use MAU. I can't say if Microsoft applications for macOS checks MAU, if present, and doesn't write/overwrite/leave those files. I can't say for certain if the MAU for Edge is specific to it and each Microsoft app comes bundled with its specific MAU. But, it was a very good point you made.

**Review by dwardu89 — APPROVED** [review]

**Comment by peterlewis** [comment]
> I didn't really consider it since I took all of a couple of minutes to submit a copy/paste. After your suggestion about MAU, then I agreed about the problem that could occur because other casks which would install and use MAU. I can't say if Microsoft applications for macOS checks MAU, if present, and doesn't write/overwrite/leave those files. I can't say for certain if the MAU for Edge is specific to it and each Microsoft app comes bundled with its specific MAU. But, it was a very good point you made.

Well thank you for taking the time to respond to the feedback, I really appreciate it.

Just for your info, MAU is MAU, i.e. it's not different, depending on the Microsoft app installed.

As much the idea of it doesn't feel me with joy, I think that including MAU in the `zap` would probably be the "right" thing to do as per the [cask language reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md) but I welcome the feedback of others in the community on this one as I don't know for sure what the correct decision would be!

It's been enjoyable contributing alongside you on cask addition! :) Thank you!

**Comment by robertleeblairjr** [comment]
> > I didn't really consider it since I took all of a couple of minutes to submit a copy/paste. After your suggestion about MAU, then I agreed about the problem that could occur because other casks which would install and use MAU. I can't say if Microsoft applications for macOS checks MAU, if present, and doesn't write/overwrite/leave those files. I can't say for certain if the MAU for Edge is specific to it and each Microsoft app comes bundled with its specific MAU. But, it was a very good point you made.
> 
> Well thank you for taking the time to respond to the feedback, I really appreciate it.
> 
> Just for your info, MAU is MAU, i.e. it's not different, depending on the Microsoft app installed.
> 
> As much the idea of it doesn't feel me with joy, I think that including MAU in the `zap` would probably be the "right" thing to do as per the [cask language reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md) but I welcome the feedback of others in the community on this one as I don't know for sure what the correct decision would be!
> 
> It's been enjoyable contributing alongside you on cask addition! :) Thank you!

Same to you and I'll look for more info in my spare time about MAU on macOS. I have VSCode installed that I played around with. It may use the same update/check methods as the Edge and other Microsoft products. If you happen to find any more relevant info, then feel free to let me know and I'll change this one. I would have thought if MAU being zapped in the previous beta or dev casks was an issue that it would have been removed but ya never know. Take care and TTYL.

**Comment by vitorgalvao** [comment]
> Based on the above, do you think that including MAU in the zap (along with the user preferences) might be the right decision, afterall?

It is, because of the rule you mentioned. If you inspect other Microsoft casks, you’ll see that’s common.

The file is missing the `.rb` at the end.

I haven’t read the whole conversation, but I’m pleased by the collaboration and politeness. Thank you both.

**Comment by robertleeblairjr** [comment]
> > Based on the above, do you think that including MAU in the zap (along with the user preferences) might be the right decision, afterall?
> 
> It is, because of the rule you mentioned. If you inspect other Microsoft casks, you’ll see that’s common.
> 
> The file is missing the `.rb` at the end.
> 
> I haven’t read the whole conversation, but I’m pleased by the collaboration and politeness. Thank you both.

Thank you. I'll add the '.rb'. Before I do; do you suggest that I add back the 'zap' for MAU.

**Comment by vitorgalvao** [comment]
> do you suggest that I add back the 'zap' for MAU.

Yes, please. `zap` should get rid of *everything* an app installs, regardless of it that source is shared.

**Review by robertleeblairjr — COMMENTED** [review]
Reverted back to reflect previously included 'zap' of MAU files.

**Comment by robertleeblairjr** [comment]
> > do you suggest that I add back the 'zap' for MAU.
> 
> Yes, please. `zap` should get rid of _everything_ an app installs, regardless of it that source is shared.

Will do. I reverted it back to include the files. Have a good day.

**Comment by reitermarkus** [comment]
This is missing the `.rb` extension.

File: Casks/microsoft-edge [hunk 5]
```diff
@@ -0,0 +1,30 @@
+cask 'microsoft-edge.rb' do
```
Comment (reitermarkus) [inline Casks/microsoft-edge]: Remove the `.rb` here.

**Comment by core-code** [comment]
sorry for being late to the party here. just wanted to ask if using this as an appcast could avoid the problem of finding the right version number for the update:

`https://macupdater.net/cgi-bin/check_urls/check_url_redirect.cgi?url=https://go.microsoft.com/fwlink/?linkid=2069148`

as it is currently the appcast only contains a version number like '80.361.20020750' which makes it hard to update the cask.

**Comment by vitorgalvao** [comment]
> sorry for being late to the party here. just wanted to ask if using this as an appcast could avoid the problem of finding the right version number for the update:
> 
> `https://macupdater.net/cgi-bin/check_urls/check_url_redirect.cgi?url=https://go.microsoft.com/fwlink/?linkid=2069148`

Yes, that does look better.

**Pull request merged**